service-base/src/main/java/org/eclipse/hono/service/registration/BaseDeviceRegistryServer.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.registration;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.healthchecks.HealthCheckHandler;
+import io.vertx.ext.healthchecks.Status;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonSession;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.connection.ConnectionFactory;
+import org.eclipse.hono.service.amqp.AmqpServiceBase;
+import org.eclipse.hono.service.amqp.Endpoint;
+import org.eclipse.hono.service.auth.AuthenticationConstants;
+import org.eclipse.hono.util.Constants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.Objects;
+import java.util.UUID;
+
+
+/**
+ * A server that implements Hono's <a href="https://www.eclipse.org/hono/api/Device-Registration-API/">Device Registration API</a> and
+ * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a> as the default implementation of Hono's device registry.
+ *
+ * TODO: add metrics.
+ * TODO: refactor this into a more generic server class (eliminate device registry specific parts).
+ */
+public class BaseDeviceRegistryServer<T extends ServiceConfigProperties> extends AmqpServiceBase<T> {
+    private ConnectionFactory authenticationService;
+
+    /**
+     * Sets the factory to use for creating an AMQP 1.0 connection to
+     * the Authentication service.
+     *
+     * @param factory The factory.
+     * @throws NullPointerException if factory is {@code null}.
+     */
+    @Autowired
+    @Qualifier(AuthenticationConstants.QUALIFIER_AUTHENTICATION)
+    public void setAuthenticationServiceConnectionFactory(final ConnectionFactory factory) {
+        authenticationService = Objects.requireNonNull(factory);
+    }
+
+    @Override
+    protected void onRemoteConnectionOpen(final ProtonConnection connection) {
+        connection.setContainer(String.format("Hono-DeviceRegistry-%s:%d", getBindAddress(), getPort()));
+        setRemoteConnectionOpenHandler(connection);
+    }
+
+    @Override
+    protected void onRemoteConnectionOpenInsecurePort(final ProtonConnection connection) {
```

Comment:
IMHO this needs a JavaDoc comment describing what this implementation actually does and that subclasses may/should override it to register other handlers etc.